src/repl.coffee

```diff
@@ -7,7 +7,8 @@ CoffeeScript = require './coffeescript'
 
 replDefaults =
   prompt: 'coffee> ',
-  historyFile: path.join process.env.HOME, '.coffee_history' if process.env.HOME
+  historyFile: do (p = process.env.XDG_CACHE_HOME or process.env.HOME) ->
+    path.join p, '.coffee_history' if p
   historyMaxInputSize: 10240
```

Comment:
I disapprove of variable assignments in function parameter lists, and of single-character variable names other than indexes/iterators. (I know, there are examples of both in the CoffeeScript codebase, but I’ve been slowly removing them as I’ve encountered them.) Also what if neither of those environment variables is set? `historyFile` should be set to _something_.

I suggest:
```coffee
historyFile: do ->
  historyPath = process.env.XDG_CACHE_HOME or process.env.HOME or process.cwd()
  path.join historyPath, '.coffee_history'
```